### DISCUSSION TITLE
Language Configuration: Add characters to new line in onEnterRules indentOutdent

### DISCUSSION THREAD

**Discussion by tim-hilt** [microsoft/vscode-discussions#418] [Extension Development QnA]
I want to write an extension that can add formatted C-style comment blocks like this:

``` c
/*
 * Hi there!
 */
```

I have gotten it to print the following, when I have `/*|*/<enter>` (with `|` being the cursor):

``` c
/*
 * |
*/
```

as you can see only the last line is off.

I wonder if I can add a single space to that last line somehow. These are my `onEnterRules` so far. For now the regexes are not tailored for anything complex.

``` json
{
  "onEnterRules": [
    {
      "beforeText": "/\\*",
      "afterText": "*/",
      "action": {
        "indent": "none",
        "appendText": " * "
      }
    },
    {
      "beforeText": "^.*\\*",
      "action": {
        "indent": "none",
        "appendText": "* "
      }
    }
  ]
}
```

https://github.com/tim-hilt/auto-comment-blocks-2/blob/3aad2c49b8a13471552635434040e86153983d72/language-configuration.json#L1-L19

Maybe `indentOutdent` isn't the right thing to use here, however it seems pretty close in doing what I actually expect it to do.

The extension is available [here](https://github.com/tim-hilt/auto-comment-blocks-2), if you want to test it.

**Comment by olguzzar** [comment]
Bumping this as it's been a while since there's been any activity - are you still having issues with this?

**Comment by tim-hilt** [comment]
I do! I just stopped pursuing this further, because it seemed impossible to do. If there would be a solution to achieve what I was looking for, I would like to give it another shot.

**Reply by burkeholland** [reply]
Closing since there hasn't been a response here. Feel free to re-open if it's still a problem.
